pkg/trace/stats/client_stats_aggregator.go

```diff
@@ -0,0 +1,283 @@
+package stats
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/watchdog"
+)
+
+const (
+	aggBucketSize           = uint64(1e9)
+	bucketDuration          = uint64(1e10) // 10s
+	oldestBucketStartSecond = uint64(20)   // 20s
+	distributionsPayload    = "distributions"
+	aggregatedCountPayload  = "counts"
+)
+
+// ClientStatsAggregator aggregates client stats payloads on 1s buckets.
+// If a single payload is received on a bucket, this Aggregator is a passthrough.
+// If two ore more payloads collide, all payloads will be sent with hits and errors
+// set to 0. An aggregation will be started for Hits/Errors/Duration counts.
+// The Aggregation will be sent through a new stat payload with no Histograms.
```

Comment:
Isn't it easier to just say:
```go
// If two or more payloads collide, they will be aggregated into the bucket duration
// and send as a new stat payload.
```
The rest of the detail is ambiguous and raises unanswered questions:
* Why are they being sent with their values set to 0?
* Why send payloads with values set to 0 but also with aggregation? Why not just the aggregation?
* etc.
Perhaps we can answer these questions later on, inside the code, inline, with documentation.